src/components/pages/Dashboard.jsx

```diff
@@ -1,33 +1,48 @@
-import { logout } from "../../util/DataBaseRequests";
+import { Outlet, useLocation } from "react-router-dom";
+import { useEffect, useState } from "react";
+
+import DashboardNav from "../navbars/DashboardNav";
+import { Home } from "./dashboard-pages/index";
+import { getUserData } from "../../util/DataBaseRequests";
 import { useAuth } from "../../AuthProvider";
-import { useNavigate } from "react-router-dom";
-import { useState } from "react";
 
 const Dashboard = () => {
-    // temporarily this functionality is here, then it will be moved to the header
-    const [logoutError, setLogoutError] = useState({});
-    const { clearUserSession, userData } = useAuth();
-    
-    const navigate = useNavigate();
-    const handleLogout = async () => {
+    const location = useLocation();
+    const [profile, setProfile] = useState({});
+    const [error, setError] = useState({});
+    const { userData, clearUserSession } = useAuth();
+
+    useEffect(() => {
+        console.log('useeffect')
+        const fetchUserProfileData = async () => {
         try {
-            const result = await logout();
-            if (result.status === 200){
-                clearUserSession();
-                navigate('/');
+            const result = await getUserData(userData._id, userData.token);
```

Comment:
Is user data not currently being set globally during login? If so accessing it from that context would eliminate another fetch request.

Reply:
The issue is that when a user logs in, we only receive part of the user's information from the backend. But in the profile we should have more information (for example, the child’s date of birth, information about the parents). Oksana and I discussed this yesterday. They will add receiving additional information at the time the user logs in. Therefore, there will be no need to get the additional information about the user.